benchmark-service/benchmark/submit.py

```diff
@@ -0,0 +1,257 @@
+        j.command(f"time tar -cf {r.name()}.tar benchmark-resources/{r.name()} --exclude='*.crc'")
+        j.command(f'ls -lh {r.name()}.tar')
+        j.command(f'mv {r.name()}.tar {j.ofile}')
+        resource_jobs[r] = j
+
+    all_benchmarks = list_benchmarks()
+    assert len(all_benchmarks) > 0
+
+    all_output = []
+
+    n_passed_filter = 0
+    job_fs = []
+    for benchmark in all_benchmarks:
+        if benchmark.name in test_names:
+            n_passed_filter += 1
+            for replicate in range(n_replicates):
+                job_fs.append((benchmark.name, replicate, benchmark.groups))
+
+    log.info(f'generating {n_passed_filter} * {n_replicates} = {n_passed_filter * n_replicates} individual benchmark jobs')
+
+    random.shuffle(job_fs)
+
+    for name, replicate, groups in job_fs:
+        j = b.new_job(name=f'{name}_{replicate}')
+        j.command(f'mv {build_hail.wheel} hail--py3-none-any.whl')
+        j.command('pip install hail--py3-none-any.whl')
+        j.command(f'mv {build_benchmark.wheel} benchmark_hail--py3-none-any.whl')
+        j.command('pip install benchmark_hail--py3-none-any.whl')
+        j.command('mkdir -p benchmark-resources')
+        for resource_group in groups:
+            resource_job = resource_jobs[resource_group]
+            j.command(f'mv {resource_job.ofile} benchmark-resources/{resource_group.name()}.tar')
+            j.command(f'time tar -xf benchmark-resources/{resource_group.name()}.tar')
+        j.command(f'MKL_NUM_THREADS=1'
+                  f'OPENBLAS_NUM_THREADS=1'
+                  f'OMP_NUM_THREADS=1'
+                  f'VECLIB_MAXIMUM_THREADS=1'
+                  f'PYSPARK_SUBMIT_ARGS="--driver-memory 6G pyspark-shell" '
+                  f'hail-bench run -o {j.ofile} -n {n_iters} --data-dir benchmark-resources -t {name}')
+        all_output.append(j.ofile)
+
+    combine_branch_factor = int(os.environ.get('BENCHMARK_BRANCH_FACTOR', 32))
+    phase_i = 1
+    while len(all_output) > combine_branch_factor:
+        new_output = []
+
+        job_i = 1
+        i = 0
+        while i < len(all_output):
+            combine = b.new_job(f'combine_output_phase{phase_i}_job{job_i}')
+            combine.command(f'mv {build_hail.wheel} hail--py3-none-any.whl')
+            combine.command('pip install hail--py3-none-any.whl')
+            combine.command(f'mv {build_benchmark.wheel} benchmark_hail--py3-none-any.whl')
+            combine.command('pip install benchmark_hail--py3-none-any.whl')
+            combine.command(
+                f'hail-bench combine -o {combine.ofile} ' + ' '.join(all_output[i:i + combine_branch_factor]))
+            new_output.append(combine.ofile)
+            i += combine_branch_factor
+            job_i += 1
+
+        phase_i += 1
+        all_output = new_output
+
+    combine = b.new_job('final_combine_output')
+    combine.command(f'mv {build_hail.wheel} hail--py3-none-any.whl')
```

Comment:
I don't think you need the double dash here

Reply:
when I remove the double dash or replace it with one dash I get errors saying it's not a valid file name

Reply:
Interesting. The name has to be a certain format. I think you can just do this `hail-0.2-py3-none-any.whl`. The version doesn't really matter since this is only being used locally.